src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -136,52 +129,43 @@ final case class PurposeApiServiceImpl(
 
   override def updatePurpose(purposeId: String, purposeUpdateContent: PurposeUpdateContent)(implicit
     contexts: Seq[(String, String)],
-    toEntityMarshallerPurpose: ToEntityMarshaller[OldPurpose],
+    toEntityMarshallerPurpose: ToEntityMarshaller[Purpose],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem]
   ): Route = authorize(ADMIN_ROLE) {
     val operationLabel = s"Updating Purpose $purposeId"
     logger.info(operationLabel)
 
-    val result: Future[OldPurpose] = for {
-      organizationId  <- getOrganizationIdFutureUUID(contexts)
-      purposeUUID     <- purposeId.toFutureUUID
-      purpose         <- purposeManagementService.getPurpose(purposeUUID)
-      ownership       <- assertOrganizationIsAConsumer(organizationId, purpose.consumerId)
-      depPayload      <- PurposeUpdateContentConverter.apiToDependency(purposeUpdateContent).toFuture
-      updatedPurpose  <- purposeManagementService.updatePurpose(purposeUUID, depPayload)
-      eService        <- catalogManagementService.getEServiceById(updatedPurpose.eserviceId)
-      enhancedPurpose <- enhancePurpose(updatedPurpose, eService, ownership)
-    } yield enhancedPurpose
-
-    onComplete(result) { updatePurposeResponse[OldPurpose](operationLabel)(updatePurpose200) }
+    val result: Future[Purpose] = for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      purposeUUID    <- purposeId.toFutureUUID
+      purpose        <- purposeManagementService.getPurpose(purposeUUID)
+      _              <- assertOrganizationIsAConsumer(organizationId, purpose.consumerId)
+      depPayload     <- PurposeUpdateContentConverter.apiToDependency(purposeUpdateContent).toFuture
+      _              <- purposeManagementService.updatePurpose(purposeUUID, depPayload)
+    } yield PurposeConverter.dependencyToApi(purpose)
```

Comment:
Otherwise the endpoint will return the old values
```suggestion
      updatedPurpose              <- purposeManagementService.updatePurpose(purposeUUID, depPayload)
    } yield PurposeConverter.dependencyToApi(updatedPurpose)
```